File: examples/documents/list-owned.swift
import JsBaoClient

// Documents the user owns (created them, or had ownership transferred to them).
func listOwnedDocuments(client: JsBaoClient) async throws {
  // #region example
  // A typed `[DocumentInfo]` — each row carries title, permission, tags, …
  let owned = try await client.me.ownedDocuments(tag: "channel")

  for doc in owned {
    print(doc.title, doc.permission)
  }
  // #endregion example
  _ = owned
}

File: examples/documents/list-owned.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Documents the user owns (created them, or had ownership transferred to them).
export async function listOwnedDocuments(client: JsBaoClient) {
  // #region example
  // Paginated page — the unified { items, cursor } envelope, same shape as
  // sharedDocuments():
  const page = await client.me.ownedDocuments({
    tag: "channel",
    returnPage: true,
  });
  const { items, cursor } = page;

  // (Without `returnPage`, the JS client returns a flat `DocumentInfo[]` for
  // convenience: `const owned = await client.me.ownedDocuments({ tag: "channel" })`.)
  // #endregion example
  return { items, cursor };
}
